tests/metrics_test.go

```diff
@@ -153,14 +159,119 @@ var _ = Describe("Metrics", func() {
 		)
 	})
 
+	Context("SSP metrics", func() {
+		var originalVersion string
+
+		getDeployment := func(name string, namespace string) *apps.Deployment {
+			deployment := &apps.Deployment{}
+			err := apiClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployment)
+			Expect(err).ToNot(HaveOccurred())
+			return deployment
+		}
+
+		totalRestoredTemplatesCount := func() int {
+			sum := 0
+			operatorPods, operatorMetricsPort := operatorPodsWithMetricsPort()
+			for _, sspOperator := range operatorPods {
+				sum += intMetricValue("total_restored_common_templates", operatorMetricsPort, &sspOperator)
+			}
+			return sum
+		}
+
+		saveOriginalVersion := func() {
+			originalVersion = ""
+
+			deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
+			envs := deployment.Spec.Template.Spec.Containers[0].Env
+
+			for _, env := range envs {
+				if env.Name == common.OperatorVersionKey {
+					originalVersion = env.Value
+				}
+			}
+		}
+
+		updateVersion := func(version string) {
+			deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
+
+			envs := deployment.Spec.Template.Spec.Containers[0].Env
+			var newEnvs []k8sv1.EnvVar
+
+			for _, env := range envs {
+				if env.Name != common.OperatorVersionKey {
+					newEnvs = append(newEnvs, env)
+				}
+			}
+
+			newEnvs = append(newEnvs, k8sv1.EnvVar{Name: common.OperatorVersionKey, Value: version})
+			deployment.Spec.Template.Spec.Containers[0].Env = newEnvs
+
+			Eventually(func() error {
+				return apiClient.Update(ctx, deployment)
+			}, 1*time.Minute, time.Second).ShouldNot(HaveOccurred())
+		}
+
+		waitForVersionUpdate := func(version string) {
+			Eventually(func() error {
+				operatorPods, _ := operatorPodsWithMetricsPort()
```

Comment:
I think you don't need to check pods here. It should be enough to test status of deployment.
These fields:
- `status.updatedReplicas` - number of running pods with the desired spec
- `status.readyReplicas` - number of pods with Ready condition

If both of these values are equal to `spec.replicas`, we know that all pods have been updated and are ready.

Reply:
does not work as is, considers the previous deployment as update and succeeds instantly

Reply:
IIUC a new ReplicaSet should be created after the update, right? What if you check the ReplicaSet for Desired == Ready Pods?

Reply:
updated